libs/ledger-live-common/src/platform/providers/RampCatalogProvider/types.ts

```diff
@@ -8,47 +10,26 @@ export type PaymentServiceProvider =
   | "applepay"
   | "googlepay";
 
-export type QueryParams = {
-  accountId?: string;
-  accountAddress?: string;
-  language?: string;
-  fiatCurrencyId?: string;
-  cryptoCurrencyId?: string;
-  primaryColor?: string;
-  mode?: string;
-  fiatAmount?: string;
-  cryptoAmount?: string;
-  address?: string;
-  theme?: string;
-};
-
-export type CryptoCurrency = {
-  id: string;
-  providerId: string;
-  ticker: string;
-};
-
-export interface GenericRampCatalogEntry {
-  name: string;
-  paramsMapping: QueryParams;
-  paymentProviders: PaymentServiceProvider[];
-  cryptoCurrencies: CryptoCurrency[];
-  fiatCurrencies: string[];
-}
-
-export interface RampLiveAppCatalogEntry extends GenericRampCatalogEntry {
-  type: "LIVE_APP";
-  appId: string;
-}
-
-export interface RampNativeCatalogEntry extends GenericRampCatalogEntry {
-  type: "NATIVE";
-  path: string;
-}
-
-export type RampCatalogEntry = RampLiveAppCatalogEntry | RampNativeCatalogEntry;
+export type CurrenciesPerProvider = Record<
+  BuyServiceProvider | string,
+  Array<CryptoCurrency["id"]>
+>;
 
 export type RampCatalog = {
-  onRamp: RampCatalogEntry[];
-  offRamp: RampCatalogEntry[];
+  onRamp: CurrenciesPerProvider;
+  offRamp: CurrenciesPerProvider;
 };
+
+export type BuyServiceProvider =
```

Comment:
This shouldn't be a closed enum actually, as we want to be able to add new providers in Buy app / Buy service without changing the LL.

I believe this type is not checked at runtime anyway, but I prefer to have explicit forward compatibility with any string allowed.